### PR TITLE
Add repo analysis notebook

### LINKED ISSUE #19: KPI metric preprocessing notebooks
Now that we have explored what the raw GitHub data looks like (#16), we should implement notebooks to analyze, preprocess and store metrics data.

**Acceptance Criteria**

- [x] Pull most relevant and updated (till current day) issue/PR data for org/repo
- [x] Preprocess data for metrics (those defined in #3)
- [x] Store the data in Ceph and create Trino tables
- [x] 1 simple visualization per metric (Superset)

### PR DESCRIPTION
**Pull request by hemajv**
Adding a repo analysis template notebook

(Resolves #19)

Co-authored-by: Hema Veeradhi <hveeradh@redhat.com>
Co-authored-by: Shrey Anand <shanand@redhat.com>
Co-authored-by: Karanraj Chauhan <kachau@redhat.com>

### CODE REVIEW

**Comment by MichaelClifford** [comment]
LGTM :) but want to give others a chance to review before approve. Good job :rocket:

File: notebooks/s3_communication.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
+"""S3 communication tools."""
```
Comment (MichaelClifford) [inline notebooks/s3_communication.py:1]: Off Topic: Is this something we should put in an OSG-utils (or some other name) package on a package index to make more easily usable?  Seems fairly generic and applicable to multiple projects. WDYT?

File: notebooks/Repo_Analysis.ipynb [hunk 2]
```diff
@@ -0,0 +1,1673 @@
+{
```
Comment (oindrillac) [inline notebooks/Repo_Analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    !python -m srcopsmetrics.cli -clr $repo_slug -e Issue,PullRequest
</pre><p>If we are planning to use the MI tool from within the notebook, do we also want to save a the raw Issue and PullRequest data on S3? I believe <code>-clr</code> only gets the data locally.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-services-group/metrics/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

**Review by aakankshaduggal — COMMENTED** [review]
Looks great :+1: 
/lgtm

**Review by suppathak — COMMENTED** [review]
very nice documentation :smile: :clap: 

/lgtm

File: notebooks/s3_communication.py (hunk 1 above)
Comment (Shreyanand) [reply notebooks/s3_communication.py:1]: Yes, we think that's a good idea. @chauhankaranraj let's set time to do this finally?

File: notebooks/Repo_Analysis.ipynb (hunk 2 above)
Comment (oindrillac) [reply notebooks/Repo_Analysis.ipynb:1]: I think the metrics table is be a transformed table, mostly a subset of the original github data, it seems to omit some columns from the raw data. One possible benefit of storing the raw data could be to facilitate future re-analysis (eg ability to add additional metrics on the data belonging to the same time frame) and reproducibility. If we think the metrics data stores all original columns that will be important to us, there is no need of the raw data.

Comment (Shreyanand) [reply notebooks/Repo_Analysis.ipynb:1]: Oh yeah I agree, the metrics data should include all the columns we are interested in. If there's additional columns in future we can add it to the same table with a one time data fetch for the past data, wdyt?

Comment (oindrillac) [reply notebooks/Repo_Analysis.ipynb:1]: sounds good!
/lgtm

**Pull request merged**